File: 1198-find-smallest-common-element-in-all-rows/1198-find-smallest-common-element-in-all-rows.java
class Solution {
    public int smallestCommonElement(int[][] mat) {
        int m = mat.length, n = mat[0].length;
        Map<Integer, Integer> counts = new HashMap<>();
        for(int i = 0; i < m; i++){
            for(int j = 0; j < n; j++){
                int count = counts.getOrDefault(mat[i][j], 0) + 1;
                if(count == m){
                    return mat[i][j];
                }
                counts.put(mat[i][j], count);
            }
        }
        return -1;
    }
}

File: 1198-find-smallest-common-element-in-all-rows/1198-find-smallest-common-element-in-all-rows.py
class Solution:
    def smallestCommonElement(self, mat: List[List[int]]) -> int:
        rset = set(mat[0])
        for i in range(1, len(mat)):
            nrset = set()
            for v in mat[i]:
                if v in rset:
                    nrset.add(v)
            rset = nrset
        return min(rset)